### DISCUSSION TITLE
How to run each test script in a separate session

### DISCUSSION THREAD

**Discussion by harupy** [pytest-dev/pytest#8072] [Q&A] [answered]
Let's say you have three test scripts:

```
- tests
  - test1.py
  - test2.py
  - test3.py
```

If you run `pytest tests`, pytest runs all the scripts in the same session.

```
> pytest tests

======== test session starts ==========
...
tests/test1.py  PASSED
tests/test2.py  PASSED
tests/test3.py  PASSED
```

Does `pytest` provide an option that allows running each test script in a separate session?

```
> pytest tests --some-option

======== test session starts ==========
...
tests/test1.py  PASSED

======== test session starts ==========
...
tests/test2.py  PASSED

======== test session starts ==========
...
tests/test3.py  PASSED
```

**Comment by RonnyPfannschmidt** [comment, marked as answer]
no

**Reply by RonnyPfannschmidt** [reply]
to elaborate - pytest has no notion of a "script" pytest has a notion of python files containing a number of test-cases

depending on usage of features like  fixtures, its even possible to have tests of different files run interspares to make use of shared setup/teardown

if your testing has the notion of a "script", then its something that pytest does not manage/is aware of

**Reply by harupy** [reply]
@RonnyPfannschmidt Thanks for the answer!
